### ISSUE TITLE
Some gossip never seems to sync in laboratory setup

### ISSUE THREAD

**Issue by joostjager** [lightningnetwork/lnd#6531]
### Background

I am experimenting with a pathfinding benchmark setup that spins up a bunch of nodes inside docker: https://github.com/bottlepay/pathfinding-benchmark/tree/lnd-6531

One essential step before starting the test is to make sure that all channels are opened and the latest channel policies gossiped to the test node that is going to make the test payments.

To detect whether the test node is fully synced, I set the default lnd base fee to 999 everywhere. What I'd want to see in the `DescribeGraph` output of the test node is every channel represented with policies that don't have a 999 base fee. (Unfortunately it seems impossible to override the policy directly when a channel is created with lnd, meaning that an extra channel_update is always needed.)

### Your environment

* lnd v0.14.3-beta
* bitcoind

### Steps to reproduce

Run `run.sh`

Note: make sure that there's sufficient memory, otherwise you may run into the death loop that is described here: https://github.com/lightningnetwork/lnd/issues/6210#issuecomment-1123736209

### Expected behaviour

At some point during the setup stage, the test runner will wait for all channel policies to be received. In the used test graph, there should be 94 non-999-basefee policies. Then it proceeds with making the pathfinding test payments.

### Actual behaviour

The test runner gets stuck on
```
testrunner_1               | 2022-05-12T14:24:12.807Z	DEBUG	Gossiped edges	{"count": 90, "expected": 94}
```

It keeps waiting for the last 4 (number varies) up to date policies, but isn't getting them.

To try to get propagation working, I'm using the following `lnd.conf` settings:

```
trickledelay=100
historicalsyncinterval=10s
```

Also the code waits for 2 minutes before changing the channel policies from the default, to avoid hitting the gossip rate limit.

Repro isn't 100%, sometimes it does get fully synced.

I've encountered propagation issues on mainnet too occasionally. Of course there could be many lnd and non-lnd related reasons for that. But perhaps this set up could help uncover a latent bug somewhere in the gossip chain.

**Comment by joostjager** [comment]
Some things that I've found out so far:
* The missing channel updates never even left the node that sets the policy. The problem isn't propagation on the network, but the update not even being broadcast.
* The reason why it isn't broadcast in`AuthenticatedGossiper.sendBatch` is because the active syncers don't have a an update horizon set (https://github.com/lightningnetwork/lnd/blob/7106ea59db9ade78d65bb46605f94637900c0e3f/discovery/syncer.go#L1250)
* I looked at that peers and they indeed never sent a `GossipTimestampRange` message to set the horizon.

I am not familiar enough with this sub-system to determine what this means exactly, but perhaps it gives a clue?

**Comment by joostjager** [comment]
@saubyk is this issue worth prioritizing? There are quite a few open issues/bugs around gossip that may be hard to reproduce. But this one is easy to repro reliably.

I know from experience that this bug has a big impact. Without a patch applied, I couldn't get my node announcement to broadcast at all.

**Comment by joostjager** [comment]
It seems that one is not _the_ fix: https://github.com/lightningnetwork/lnd/pull/7186#issuecomment-1338012722

**Comment by yyforyongyu** [comment]
Nope it's not. Meanwhile what's the topology looking like? I'll try to run the experiment to see what went wrong.

**Comment by joostjager** [comment]
The topology is defined [here](https://github.com/bottlepay/pathfinding-benchmark/blob/master/graph.yml). It are a lot of nodes, and part of the test setup is to verify that the designated sender node has received all channel updates.

For this test, I use a modified lnd version: https://github.com/bottlepay/lnd/commits/pathfinding-benchmark-mod-mpp

It contains three changes:
* Reduced scrypt parameters to improve memory footprint
* Reduced cache pre-allocation to improve memory footprint
* The [gossip patch](https://github.com/bottlepay/lnd/commit/41564ff4588b62bc517b85f9adad05f8124a5817).

Without the patch, the sender node isn't going to get a full picture of the graph.

**Comment by joostjager** [comment]
@Roasbeef ran it three times to be sure, and problem seems solved wtih #7239
